crypto/fipsmodule/sha/sha3_test.cc

```diff
@@ -67,6 +115,195 @@ class SHA3TestVector {
               Bytes(digest_.data(), digest_length));
   }
 
+  // Test SHAKE Squeeze functionality through |EVP_Digest| APIs
+  void NISTTestVectors_SHAKESqueeze(const EVP_MD *algorithm) const {
+    size_t sqd_bytes = 0, cur_test = 0, to_sq_bytes = 0;
+
+    uint32_t digest_length = out_len_ / 8;
+    std::unique_ptr<uint8_t[]> digest(new uint8_t[digest_length]);
+    bssl::ScopedEVP_MD_CTX ctx;
+
+    // Test Final XOF
+    // Assert fail when |EVP_DigestFinalXOF| is called as a streaming API
+    ASSERT_TRUE(EVP_DigestInit(ctx.get(), algorithm));
+    ASSERT_TRUE(EVP_DigestUpdate(ctx.get(), msg_.data(), msg_.size()));
+    ASSERT_TRUE(EVP_DigestFinalXOF(ctx.get(), digest.get(), digest_length));
+    EXPECT_EQ(Bytes(digest.get(), digest_length),
+              Bytes(digest_.data(), digest_length));
+
+    ASSERT_FALSE(EVP_DigestFinalXOF(ctx.get(), digest.get(), digest_length));
+    ASSERT_FALSE(EVP_DigestSqueeze(ctx.get(), digest.get(), digest_length));
+
+    // Test the one-shot
+    // Assert success when |EVP_Digest| is called
+    OPENSSL_memset(digest.get(), 0, digest_length);
+    ASSERT_TRUE(EVP_Digest(msg_.data(), msg_.size(), digest.get(),
+                           &digest_length, algorithm, nullptr));
+    EXPECT_EQ(Bytes(digest.get(), digest_length),
+              Bytes(digest_.data(), digest_length));
+
+    // Test Final
+    // Assert fail when |EVP_DigestFinal| is called for XOF algorithms
+    OPENSSL_memset(digest.get(), 0, digest_length);
+    ASSERT_TRUE(EVP_DigestInit(ctx.get(), algorithm));
+    ASSERT_TRUE(EVP_DigestUpdate(ctx.get(), msg_.data(), msg_.size()));
+
+    ASSERT_FALSE(EVP_DigestFinal(ctx.get(), digest.get(), &digest_length));
+    ASSERT_FALSE(EVP_DigestFinalXOF(ctx.get(), digest.get(), digest_length));
```

Comment:
Interesting, even though the `EVP_DigestFinal` call failed it has corrupted the ctx somehow such that `EVP_DigestFinalXOF` also fails now?